### PR TITLE
chore(ci): Revert upload-artifact version

### PR DESCRIPTION
**Pull request by azaslavsky**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -41,7 +41,7 @@ jobs:
           mv target/release/symsorter build/symsorter-Linux-x86_64
           mv target/release/symbolicli build/symbolicli-Linux-x86_64
 
-      - uses: actions/upload-artifact@v4.3.1
+      - uses: actions/upload-artifact@v3
```
Comment (hubertdeng123) [inline .github/workflows/build.yml:44]: Looks like this was v3.1.1 [here](https://github.com/getsentry/symbolicator/pull/1366/files), let's keep it the same version as before?

**Comment by hubertdeng123** [comment]
To give context, this change is needed since it's breaking the self-hosted release. There is a breaking change in upload-artifact v4 here with uploading artifacts with the same name
https://github.com/actions/toolkit/tree/main/packages/artifact

**Review by hubertdeng123 — APPROVED** [review]
lgtm pending comment on maintaing version 3.1.1

File: .github/workflows/build.yml (hunk 1 above)
Comment (azaslavsky) [reply .github/workflows/build.yml]: Done.

**Pull request merged**